### PR TITLE
server: migrate test-infra to testify for tidb_test.go

### LINKED ISSUE #27806: migrate test-infra to testify for `server/tidb_test.go`
This is a subtask of #26863

### PR DESCRIPTION
**Pull request by yedamao**
### What problem does this PR solve?

Issue Number: close #27806 #27805

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by yedamao** [comment]
@tisonkun Please take a look.

File: server/main_test.go [hunk 1]
```diff
@@ -30,12 +31,16 @@ import (
 func TestMain(m *testing.M) {
 	testbridge.WorkaroundGoCheckFlags()
 
+	runInGoTest = true // flag for NewServer to known it is running in test environment
```
Comment (tisonkun) [inline server/main_test.go:34]: Nice work!

**Review by tisonkun — APPROVED** [review]
Thanks for your contribution!

**Review by morgo — APPROVED** [review]

File: server/tidb_test.go [hunk 2]
```diff
@@ -32,102 +32,48 @@ import (
 	"os"
 	"path/filepath"
```
Comment (hawkingrei) [inline server/tidb_test.go:33]: tidb_test contains build flags.

```
//go:build !race
// +build !race
```

it need to add ```tidb_serial_test.go```

File: testkit/dbtestkit.go [hunk 3]
```diff
@@ -60,6 +83,14 @@ func (tk *DBTestKit) MustQuery(sql string, args ...interface{}) *sql.Rows {
 	return rows
 }
 
+// MustQueryRows query the statements
+func (tk *DBTestKit) MustQueryRows(query string, args ...interface{}) {
+	rows := tk.MustQuery(query, args...)
+	tk.require.True(rows.Next())
+	tk.require.NoError(rows.Err())
+	rows.Close()
```
Comment (tisonkun) [inline testkit/dbtestkit.go:91]: You should check the error return here I think. @yedamao could you submit a follow up to fix it?

**Pull request merged**